### PR TITLE
normalize UI component font styles

### PR DESCRIPTION
**Pull request by brad-decker**
Depends on #9693 

Breaking apart #9074 so that it allows for easier review in parts. #9074 was reviewed a few times and changes were made but the latest rebase puts it into a state where the full review is once again needed. This part of the effort focuses on UI components. This PR has the broadest implications given that it touches the UI library which is used throughout the site.

### CODE REVIEW

File: ui/app/components/ui/button/buttons.scss [hunk 1]
```diff
@@ -228,15 +227,6 @@ input[type="submit"][disabled] {
   opacity: 0.5;
 }
 
-button.primary {
```
Comment (brad-decker) [inline ui/app/components/ui/button/buttons.scss:231]: this is unused in the codebase and is thus removed.

File: ui/app/components/ui/loading-screen/index.scss [hunk 2]
```diff
@@ -26,13 +26,6 @@
     align-items: center;
   }
 
-  &__message {
```
Comment (brad-decker) [inline ui/app/components/ui/loading-screen/index.scss:29]: This was unused and thus removed

File: ui/app/components/ui/page-container/index.scss [hunk 3]
```diff
@@ -91,24 +92,24 @@
...
-    font-weight: 400;
   }
 
   &__title {
+    @include H2;
+
     color: $black;
-    font-size: 2rem;
     font-weight: 500;
-    line-height: 2rem;
```
Comment (brad-decker) [inline ui/app/components/ui/page-container/index.scss:104]: the resulting line height is much higher, as noted by @Gudahtt in #9074, but on review:

> The `line-height` is now much higher with this change (it went from 32px to 44.8px), but I do think this looks better in all cases I know of. I can't find any with stray padding that is now redundant.


https://github.com/MetaMask/metamask-extension/pull/9074/files?file-filters%5B%5D=.scss#r466411684

File: ui/app/components/ui/page-container/index.scss [hunk 4]
```diff
@@ -91,24 +92,24 @@
...
     margin-right: 1.5rem;
   }
 
   &__subtitle {
+    @include H6;
```
Comment (brad-decker) [inline ui/app/components/ui/page-container/index.scss:110]: this results in a small change to font size. @Gudahtt also reviewed this change here:

https://github.com/MetaMask/metamask-extension/pull/9074/files?file-filters%5B%5D=.scss#r466433436

**Review by darkwing — DISMISSED** [review]
These changes all make sense and I'm very happy that we can properly structure font sizes -- yay!

One thing that does concern me is the number of "headings" we have; `H9` feels like a sight for sore eyes.  I'd love if we could get down to less overall sizes and have a more uniform look and feel.

**Comment by brad-decker** [comment]
@darkwing -- although they are labeled as H7-H9, they are not used as headings (typically 🙃). I have also asked the design team to consider alternative, descriptive labels for these font settings, such as "subtitle," "label," etc... My experience comes from material design typography settings, which have *more* overall variations in font size but with a distinct purpose. Cc @MetaMask/design. In the previous PR that this was split from, @rachelcope also pointed out many areas where we are using the entirely wrong font settings. Still, this PR's goal was to improve visibility of which typography styles are in use. 

I hope that helps for additional context, and thanks for the review!

**Comment by brad-decker** [comment]
adding @Gudahtt as a required reviewer on all of these as he had already done an extensive review on the massive PR.

/sparkle @Gudahtt for his time regression testing my work

File: ui/app/components/ui/button-group/index.scss [hunk 5]
```diff
@@ -4,7 +4,8 @@
   align-items: center;
 
   &__button {
-    font-size: 1rem;
+    @include H5;
```
Comment (Gudahtt) [inline ui/app/components/ui/button-group/index.scss:7]: Was this issue resolved? https://github.com/MetaMask/metamask-extension/pull/9074/files#r465984022

Comment (brad-decker) [reply ui/app/components/ui/button-group/index.scss:7]: I don't think so!

Comment (brad-decker) [reply ui/app/components/ui/button-group/index.scss]: Fixed in the latest commit

File: ui/app/components/ui/unit-input/index.scss [hunk 6]
```diff
@@ -48,14 +49,14 @@
 
   &__input-container {
     display: flex;
-    align-items: flex-start;
+    align-items: center;
     padding-bottom: 4px;
```
Comment (Gudahtt) [inline ui/app/components/ui/unit-input/index.scss:53]: I had suggested in the last PR that this be reduced to `2px`, to more closely match the old height of this input (https://github.com/MetaMask/metamask-extension/pull/9074/files#r466403886)

I think that suggestion still has merit? :thinking: The difference is fairly small though.

**Review by Gudahtt — DISMISSED** [review]
LGTM!

File: ui/app/components/ui/unit-input/index.scss (hunk 6 above)
Comment (brad-decker) [reply ui/app/components/ui/unit-input/index.scss]: Done :D

**Review by Gudahtt — DISMISSED** [review]

**Comment by Gudahtt** [comment]
Though I did notice that this seems to require a rebase! The first commit is a duplicate of the one from the branch this was pointed at. I bet the other four need to be rebased as well.

**Comment by brad-decker** [comment]
@Gudahtt good catch, rebased.

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**